### PR TITLE
5/decryption with shamir shares

### PR DESCRIPTION
**Pull request by echevrier**
cargo clippy

Add function to write one line to a file. Rename into writelines/readlines the method to write/read Vec in/from lines

Add function to create a file to save the nft urllist

Add changes requested in review of PR #34

Replace url_storage with local_file_storage. Remove obsolete url_storage

Remove import of deleted decrypt

Changes required in PR #34

Local_storage_handler as trait, unit test for local storage and decrypt with shamir

commit to check out another branch

decryption

4/keyvault interaction (#30)

* add keyvault provision

* add list commands

* add url storage handler

* add list command

* cargo fmt

* add create handler

* gitignore

* rename keyvaultlist.txt to keyvault_pool.txt

* clean up get_nft_id cmds

* [UrlStorageHanlder] rename create fn to open fn

* initial provision function commit

* refactor url storage handler

* add shamir secret tests

* connect clap provision with provion impl

* add get and check impl

* remove obsolete TODO

* apply review comments

* remove clippy warning

* update unit tests

cargo fmt (#33)

Implement nft pallet interaction #3

* Create NFT functionality

* Forgotten in last commit: files for new modules: create and ternoa_implementation

* Start test mock

* Add nft transfer command.

* Add nft mutate command.

* 3/pallet interaction: add commands
Note: series_id: hardcoded to 1u32 -> create works only for one owner

* series_id: hardcoded to 0u32 (default serie), so creation of NFT works for multiple owners.
Update comments and run cargo fmt

* rebased on develop, run cargo fmt

* Catch ternoa chain error as ExtrinsicFailed in the NFT interactions and break infinite loop

* clean info message for NFT

* Add TODO according to requested changes

Co-authored-by: echevrier <edith.chevrier@scs.ch>

rebased on branch 2/aes256... version e86b85e and cargo fmt

cargo fmt

Local_storage_handler as trait, unit test for local storage and decrypt with shamir

commit to check out another branch

decryption

### CODE REVIEW

File: client/src/ternoa_implementation/local_storage_handler.rs [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+}
+
+impl VecToLinesConverter<String> for LocalFileStorage {
+    fn write(&self, line: String) -> Result<()> {
+        self.ensure_dir_exists()?;
+        let mut file = fs::File::create(&self.filepath())?;
+        file.write_all(line.as_bytes())
+    }
+
+    fn writelines(&self, lines: Vec<String>) -> Result<()> {
```
Comment (haerdib) [inline client/src/ternoa_implementation/local_storage_handler.rs:54]: why not `write_lines` ?

File: client/src/ternoa_implementation/local_storage_handler.rs [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        self.ensure_dir_exists()?;
+
+        let mut text: String = lines.iter().map(|url| format!("{}\n", url)).collect();
+        text.pop(); // remove last line break
+        let mut file = fs::File::create(&self.filepath())?;
+        file.write_all(text.as_bytes())
+    }
+
+    fn readlines(&self) -> Result<Vec<String>> {
```
Comment (haerdib) [inline client/src/ternoa_implementation/local_storage_handler.rs:63]: ^^^^^
would enhance readability

File: client/src/ternoa_implementation/keyvault/provision.rs [hunk 3]
```diff
@@ -41,15 +46,29 @@ pub fn provision(
...
+        //nft_urls.push()
     }
 
-    // TODO: create file NFT urllist NFT File
+    // Create file NFT urllist NFT File
+    save_nft_urls(nft_urls, _nft_id)?;
     Ok(())
 }
 
+fn save_nft_urls(nft_urls: Vec<String>, _nft_id: NFTId) -> Result<(), String> {
```
Comment (haerdib) [inline client/src/ternoa_implementation/keyvault/provision.rs:61]: don't use _ if the variable is used. i.e. change  `_nft_id` to `nft_id`

**Review by haerdib — APPROVED** [review]
Two minor comments. Would be gread if you resolve them before merging, but it's not that important, hence I'm approving.

File: client/src/ternoa_implementation/local_storage_handler.rs (hunk 1 above)
Comment (echevrier) [reply client/src/ternoa_implementation/local_storage_handler.rs:54]: In python its writeline ;). I will rename it

**Pull request merged**